api.go

```diff
@@ -324,11 +325,20 @@ func (api *API) ImportRoaring(ctx context.Context, indexName, fieldName string,
 					}
 					// must make a copy of data to operate on locally.
 					// field.importRoaring changes data
-					data := make([]byte, len(viewData))
-					copy(data, viewData)
-					err = field.importRoaring(data, shard, viewName, req.Clear)
-					if err != nil {
-						return err
+					fileMagic := uint32(binary.LittleEndian.Uint16(viewData[0:2]))
+					if fileMagic == roaring.MagicNumber { // if pilosa roaring
+						err = field.importRoaring(viewData, shard, viewName, req.Clear)
+						if err != nil {
+							return err
+						}
+
+					} else {
+						data := make([]byte, len(viewData))
```

Comment:
let's move the comment from above down here and edit it a bit to explain what modifies the data

Reply:
any benefit in wrapping the errors too?

Reply:
definitely... good call